code/modules/mob/living/carbon/carbon.dm

```diff
@@ -1190,3 +1190,24 @@
   */
 /mob/living/carbon/proc/get_biological_state()
 	return BIO_FLESH_BONE
+
+/// Returns whether or not the carbon should be able to be shocked
+/mob/living/carbon/proc/should_electrocute(power_source)
+	if (ismecha(loc))
+		return FALSE
+
+	if (wearing_shock_proof_gloves())
+		return FALSE
+
+	if(!get_powernet_info_from_source(power_source))
+		return FALSE
+
+	return TRUE
+
+/// Returns if the carbon is wearing shock proof gloves
+/mob/living/carbon/proc/wearing_shock_proof_gloves()
+	if (gloves)
+		var/obj/item/clothing/gloves/G = gloves
+		return G.siemens_coefficient == 0
+
+	return FALSE
```

Comment:
Do you take into account things such as Genetics Insulated and Nerve Grounding surgery as part of any of these checks?